### PR TITLE
Return transaction error to be checked for system chunk transaction

### PR DESCRIPTION
**Pull request by kc1116**
Return the system chunk transaction execution error so that it can be logged.

### CODE REVIEW

File: engine/execution/computation/computer/computer.go [hunk 1]
```diff
@@ -306,7 +314,7 @@ func (e *blockComputer) executeTransaction(
 	collectionIndex int,
 	txIndex uint32,
 	res *execution.ComputationResult,
-) error {
+) (error, txerror error) {
```
Comment (jordanschalm) [inline engine/execution/computation/computer/computer.go:317]:
```suggestion
) (executionErr error, txErr error) {
```
pretty sure go will want you to name both return values

File: engine/execution/computation/computer/computer.go [hunk 2]
```diff
@@ -239,11 +239,19 @@ func (e *blockComputer) executeSystemCollection(
...
 
-	err = e.executeTransaction(tx, colSpan, collectionView, programs, systemChunkCtx, collectionIndex, txIndex, res)
+	err, txerr := e.executeTransaction(tx, colSpan, collectionView, programs, systemChunkCtx, collectionIndex, txIndex, res)
 	txIndex++
 	if err != nil {
+		return txIndex, err
+	} else if txerr != nil {
+		e.log.Err(txerr).
+			Hex("block_id", logging.Entity(systemChunkCtx.BlockHeader)).
+			Str("system_chunk_error", "true").
```
Comment (ramtinms) [inline engine/execution/computation/computer/computer.go:249]: We might use something more generic like "critical_error": "true", so we can add it to other places as well.

Comment (jordanschalm) [inline engine/execution/computation/computer/computer.go:249]: I'd add a comment here that this log is used as the data source for an alert on grafana and that it's important the text of `system_chunk_error` field isn't changed

**Review by ramtinms — APPROVED** [review]
looks good to me,

Only flag is: 
I assume we want this to be here for some time, what if we merge this into master first and then cherry-pick it, this way we make sure all the integration tests are passing and won't forget to backport this.

**Comment by jordanschalm** [comment]
> I assume we want this to be here for some time, what if we merge this into master first and then cherry-pick it, this way we make sure all the integration tests are passing and won't forget to backport this.

👍  but we've been doing it in the opposite direction, keeping track of things that need to be backported to `master` in https://github.com/onflow/flow-go/pull/1088

File: engine/execution/computation/computer/computer.go [hunk 3]
```diff
@@ -239,11 +239,19 @@ func (e *blockComputer) executeSystemCollection(
 		return txIndex, fmt.Errorf("could not get system chunk transaction: %w", err)
 	}
 
-	err = e.executeTransaction(tx, colSpan, collectionView, programs, systemChunkCtx, collectionIndex, txIndex, res)
+	err, txerr := e.executeTransaction(tx, colSpan, collectionView, programs, systemChunkCtx, collectionIndex, txIndex, res)
```
Comment (m4ksio) [inline engine/execution/computation/computer/computer.go:242]: I would oppose this change. 
Transaction error is not an error in terms of execution - its successful execution from EN code point of view. We leak the logic here.
Tx result is available for inspection inside `res` here.

**Review by m4ksio — CHANGES_REQUESTED** [review]
Temporarily blocking to make sure we align on the method

File: engine/execution/computation/computer/computer.go (hunk 3 above)
Comment (ramtinms) [reply engine/execution/computation/computer/computer.go:242]: I misled them. I was worried that looking at the last item in res might not be trivial and would be based on assumption that this tx is always the last one, so though wiring the txError upward might be okey.

**Review by jordanschalm — APPROVED** [review]

File: engine/execution/computation/computer/computer.go [hunk 4]
```diff
@@ -241,9 +241,22 @@ func (e *blockComputer) executeSystemCollection(
...
 	err = e.executeTransaction(tx, colSpan, collectionView, programs, systemChunkCtx, collectionIndex, txIndex, res)
 	txIndex++
+
 	if err != nil {
+		return txIndex, err
+	} else if res.TransactionResults[txIndex].ErrorMessage != ""{
+		// This log is used as the data source for an alert on grafana.
+		// The system_chunk_error field must not be changed without adding the corresponding
+		// changes in grafana. https://github.com/dapperlabs/flow-internal/issues/1546
+		e.log.Err(fmt.Errorf(res.TransactionResults[txIndex].ErrorMessage)).
```
Comment (m4ksio) [inline engine/execution/computation/computer/computer.go:251]: We use `Errorf` with an input which comes from external source. This is potentially exploitable and simply unnecessary.
No need to use `Err` only to use `fmt.Errorf` to wrap a string.

File: engine/execution/computation/computer/computer.go [hunk 5]
```diff
@@ -241,9 +241,22 @@ func (e *blockComputer) executeSystemCollection(
...
+			Hex("block_id", logging.Entity(systemChunkCtx.BlockHeader)).
+			Str("system_chunk_error", "true").
+			Str("critical_error", "true").
+			Msgf("error executing system chunk transaction")
```
Comment (m4ksio) [inline engine/execution/computation/computer/computer.go:255]: No need to use `Msgf` here as well

**Review by m4ksio — CHANGES_REQUESTED** [review]
Requesting changes because of theoretical potential abuse of `Errorf``

**Pull request merged**